### ISSUE TITLE
Unable to locate executable file: godot.

### ISSUE THREAD

**Issue by knightofiam** [firebelley/godot-export#43]
Here's the relevant part of my yaml:

```
on:
  push:
    branches:
      - master

jobs:
  export:
    name: export executables
    runs-on: ubuntu-latest
    steps:
      - name: checkout code
        uses: actions/checkout@v2.3.4
        with:
          fetch-depth: 0
      - name: create executables
        uses: firebelley/godot-export@v2.6.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GODOT_MONO_VERSION: 3.2.2
        with:
          godot_executable_download_url: https://downloads.tuxfamily.org/godotengine/${{env.GODOT_MONO_VERSION}}/mono/Godot_v${{env.GODOT_MONO_VERSION}}-stable_mono_linux_headless_64.zip
          godot_export_templates_download_url: https://downloads.tuxfamily.org/godotengine/${{env.GODOT_MONO_VERSION}}/mono/Godot_v${{env.GODOT_MONO_VERSION}}-stable_mono_export_templates.tpz
          relative_project_path: ./
          use_preset_export_path: true
          archive_export_output: true
          create_release: true
          base_version: 0.0.1
```

Here's the relevant part of the log:

```
  env:
    GITHUB_TOKEN: ***
    GODOT_MONO_VERSION: 3.2.2
Download Godot
  Working path created /home/runner/.local/share/godot
  Downloading Godot export templates from https://downloads.tuxfamily.org/godotengine/3.2.2/mono/Godot_v3.2.2-stable_mono_export_templates.tpz
  Downloading Godot executable from https://downloads.tuxfamily.org/godotengine/3.2.2/mono/Godot_v3.2.2-stable_mono_linux_headless_64.zip
  /usr/bin/wget -nv https://downloads.tuxfamily.org/godotengine/3.2.2/mono/Godot_v3.2.2-stable_mono_export_templates.tpz -O /home/runner/.local/share/godot/godot_templates.tpz
  /usr/bin/wget -nv https://downloads.tuxfamily.org/godotengine/3.2.2/mono/Godot_v3.2.2-stable_mono_linux_headless_64.zip -O /home/runner/.local/share/godot/godot.zip
  2020-12-02 06:25:34 URL:https://downloads.tuxfamily.org/godotengine/3.2.2/mono/Godot_v3.2.2-stable_mono_export_templates.tpz [623608361/623608361] -> "/home/runner/.local/share/godot/godot_templates.tpz" [1]
  Error: Unable to locate executable file: godot. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

The environment-variable-injected URL's are correct because I can download the files directly from them in the log.

**Comment by knightofiam** [comment]
It's working now, although I hadn't changed anything at the time in the build configuration. It may have been a bug in a Godot Mono release that has recently been fixed, although I couldn't find anything relevant in the recent issues or commit history.

**Comment by firebelley** [comment]
This may have been a transient GitHub actions-related issue as well. I've found that sometimes actions will fail for random reasons. Still it's good to keep this issue open in case you or anyone else encounters the same thing.

**Comment by firebelley** [comment]
I am going to close this, as I haven't seen any more reports of this issue occurring. Please don't hesitate to open another issue if this happens again!

**Comment by coelhucas** [comment]
I started having the very same issue after copying my action from one project to another, even though both points to the very same binaries. I thought it might be because of the project being empty, but I added some contents to it and it still errors.

> Unable to locate executable file: godot. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.

**Comment by knightofiam** [comment]
@coelhucas Can you get the Godot download URL’s from the build log, as I demonstrated above? If you can, try copying and pasting them into the address bar in your browser and see if it starts downloading. That’s where I’d start troubleshooting.

**Comment by coelhucas** [comment]
So, here's the thing! I was working on a fork of mine and the issue happened based on whether I had or not an arbitrary line in the code. I discovered it was a small race condition happening on a `Promise.all` call.

Since they run in parallel and `prepareTemplates` depends on `prepareExecutable`, it was possible to `prepareTemplates` ask for Godot's version before the executable is ready.

I fixed it on my fork by making [these calls serial](https://github.com/softwoolco/godot-export/commit/9f6ac3957964ea96037535910be981692fd93977#diff-3d2b59189eeedc2d428ddd632e97658fe310f587f7cb63b01f9b98ffc11c0197R4282-R4283)

Since my fork have other fixes (it's meant to be used with [Godot Steam](https://github.com/Gramps/GodotSteam)) I won't open a PR right now, but if you folks think it makes sense I can open from a separate fork.
